tools/main-transpile-wrapper.js

```diff
@@ -0,0 +1,15 @@
+// This file exists because it is the file in the tool that is not automatically
+// transpiled by Babel
+
+require('babel-core/register')(require('./babel-config.js')); // #RemoveInProd this line is removed in isopack.js
```

Comment:
OK, here's a tricky thing.

babel-core/register calls sourceMapSupport.install itself: https://github.com/babel/babel/blob/050bcec617117e21e240858edf310ff25bfa8d95/src/babel/api/register/node.js#L11-L24

This happens *before* your source-map-retriever-stack calls it.

What does `install()` actually do?  https://github.com/evanw/node-source-map-support/blob/e0e8653cc85008d265051641dd82865d29fa5f83/source-map-support.js#L376-L378

It runs if and only if a `var` in that file hasn't been set yet.  If it decides it's the first execution, it overrides the global value `Error.prepareStackTrace`.

So the goal of source-map-support is to ensure that only the first call to `install()` does anything.  And yet... the code you wrote actually works (right?), which is to say that the *second* call (the one in `source-map-retriever-stack`) is taking effect.

Why's that?  Well, looks like babel-core depends on `"source-map-support": "^0.2.10"` (ie, "something compatible with 0.2.10"), and our top-level dependency is on `source-map-support: "0.3.1"`. So these are definitely going to resolve to two different copies of `source-map-support`.  (Even if we ran npm dedup or whatever!) And so the `alreadyInstalled` var is going to be different between the two, and so it's going to let you override the already overloaded `Error.prepareStackTrace`.

What's my point?  Well, right now this does work.  But if babel-core changed their dependency to something where npm was suddenly able to de-dup the two uses of the package, then it might suddenly stop working because the second call wouldn't work.

(This is arguably a bug in `source-map-support` ... since it is making a decision about overriding the global `Error.prepareStackTrace`, it should be basing its decisions on global data.  eg, maybe it should be checking `if (!(Error.prepareStackTrace && Error.prepareStackTrace._sourceMapSupport))` and setting that flag on the function when it assigns it.  Maybe should open an issue there?  Of course, making this fix would make it so your code never works instead of sometimes works, but that's arguably an improvement.)

Fixes I can imagine include "not using babel-core/register at all but writing your own similar thing" or "installing babel-core from a fork which deletes the source-map-support line".  Maybe there are others.